MusicNotationKit/MusicNotationKit/Note.swift

```diff
@@ -53,3 +53,32 @@ extension Note: NoteCollection {
 	
 	internal var noteCount: Int { return 1 }
 }
+
+extension Note: CustomDebugStringConvertible {
+	public var debugDescription: String {
+		
+		// Tones are handled in two different ways. Single notes
+		// do not contain collection notation, whereas chords do.
+		var tones = ""
+		if self.tones.count == 1 {
+			tones = String(self.tones[0])
+		} else if self.tones.count > 1 {
+			tones = String(self.tones)
+		}
+		
+		// The dot notation is placed after the note duration.
+		var duration  = String(noteDuration)
+		if dot != nil {
+			duration += String(dot!)
+		}
+		
+		var result = "\(duration)\(tones)"
+		
+		if isRest {
+			result += "R"
+		}
+		
+		
+		return result
+	}
```

Comment:
Guidelines to remember:
1. Avoid var wherever possible and use let. 
2. Never use `!`. You can always use optional unwrapping (`if-let`) and chaining (`variable?`)
3. Avoid subscripting if possible because it can crash. In this case, you `.first`.
4. Only use `self.` when necessary

Rewrite as follows:

```swift
let tones = tones.count == 1 ? String(tones.first) : String(tones)
let duration = "\(noteDuration)\(dot ?? "")"
return "\(duration)\(tones)\(isRest ? "R" : "")"
```